src/VisualStudio/Core/Def/Implementation/FindReferences/Contexts/AbstractTableDataSourceFindUsagesContext.cs

```diff
@@ -96,12 +101,46 @@ private abstract class AbstractTableDataSourceFindUsagesContext :
 
                 Debug.Assert(_findReferencesWindow.Manager.Sources.Count == 0);
 
+                var customColumnNames = customColumns.Select(c => c.Name).ToImmutableHashSet();
+
                 // And add ourselves as the source of results for the window.
-                _findReferencesWindow.Manager.AddSource(this);
+                // Additionally, add custom columns to display custom reference information.
+                _findReferencesWindow.Manager.AddSource(this, customColumnNames);
 
                 // After adding us as the source, the manager should immediately call into us to
                 // tell us what the data sink is.
                 Debug.Assert(_tableDataSink != null);
+
+                // Initialize and update custom column states.
+                _customColumnStatesMap = GetInitialCustomColumnStates(findReferencesWindow.TableControl.ColumnStates, customColumns, customColumnNames);
```

Comment:
so mostly work is just adding this custom column to table control. so, review from editor team will be helpful @olegtk

Reply:
Yes. I also need their help for the TODO (2) mentioned in [PR description](https://github.com/dotnet/roslyn/pull/30155#issue-218456999). Let me start an internal thread with them.